### DISCUSSION TITLE
When will Github Copilot close?

### DISCUSSION THREAD

**Discussion by 2br-2b** [community/community#8324] [Copilot Conversations] [answered]
Hello!

I'd like to start off by saying that Github Copilot is amazing! I think it wouldn't be too much of an overshot to say that Copilot revolutionizes how I personally write code!

Inspired by an episode of the WAN show, I am working on [a scientific study testing how Copilot will effect the academic setting](https://github.com/2br-2b/Github-Copilot-Project/). I am doing a smaller version of this study for my junior seminar this semester and am planning on expanding it for my senior thesis in the Spring of 2023.

According to https://copilot.github.com/#faq-will-there-be-a-paid-version, however, Copilot will eventually have a paid version. I don't know if this means that the current version will be shut down before that time, and while I assume that some form of Copilot will still be available, I want to make sure I will still be able to run this study.

Should I expect some form of Copilot to still be online in Spring of 2023, or should I run the study as soon as possible before Copilot is discontinued?

Thank you and God bless!

**Comment by ryanjsalva** [comment, marked as answer]
Hi, @2br-2b. Your study sounds very cool. I can’t wait to see the results! 

While we haven’t set an exact data for General Availability – and, therefore, a paid product – the Spring of 2023 is over a year way. I certainly hope we’re able to move Copilot out of preview and offer it as a fully supported product before then. If you can, I recommend running your study this Spring (2022).

**Reply by 2br-2b** [reply]
Thank you! I'll speak with an advisor and see if we can accelerate the process!
